### PR TITLE
Split out CommandLine enum into a separate static library, allow removing it from stdlib

### PR DESCRIPTION
**Pull request by kubamracek**
This is for the 'freestanding' build to stop assuming the platform has argc/argv.

- Introduce a new sub-library, libswiftCommandLineSupport.a
- Move stubs/CommandLine.cpp into this library
- Conditionally embed it into libswiftCore
- Conditionally embed it into libswiftPrivateLibcExtras if not in libswiftCore to support testing
- Add SWIFT_STDLIB_HAS_COMMANDLINE CMake (and build-script) flag

### CODE REVIEW

**Comment by kubamracek** [comment]
@swift-ci please test

**Comment by kubamracek** [comment]
@swift-ci please test

**Comment by kubamracek** [comment]
@swift-ci please test

**Comment by kubamracek** [comment]
preset=stdlib_S_standalone_minimal_macho_x86_64,build,test
@swift-ci please test with toolchain and preset

File: lib/Immediate/Immediate.cpp [hunk 1]
```diff
@@ -238,7 +238,7 @@ int swift::RunImmediately(CompilerInstance &CI,
   // This must be done here, before any library loading has been done, to avoid
   // racing with the static initializers in user code.
   // Setup interpreted process arguments.
-  using ArgOverride = void (*)(const char **, int);
+  using ArgOverride = void (* SWIFT_CC(swift))(const char **, int);
```
Comment (compnerd) [inline lib/Immediate/Immediate.cpp:241]: Nice find - I suspect that the WASM folks would  appreciate this too

**Review by compnerd — COMMENTED** [review]
I don't see anything worrisome except for the potential ABI issues.  I'd like to get @mikeash and @jckarter to have a look over this as well.

**Review by mikeash — APPROVED** [review]

**Comment by mikeash** [comment]
Oh right. I somehow got it in my head that the CommandLine enum was in the static library. You’re right, no module names at play here, and it’s all good.

**Comment by kubamracek** [comment]
preset=stdlib_S_standalone_minimal_macho_x86_64,build,test
@swift-ci please test with toolchain and preset

**Pull request merged**